motionplan/constraint_test.go

```diff
@@ -333,3 +334,42 @@ func BenchmarkCollisionConstraints(b *testing.B) {
 	}
 	bt = b1
 }
+
+func TestConstraintConstructors(t *testing.T) {
+	c := CreateConstraints()
+
+	desiredLinearTolerance := float32(1000.0)
+	desiredOrientationTolerance := float32(0.0)
+
+	AddLinearConstraint(c, &desiredLinearTolerance, &desiredOrientationTolerance)
+	test.That(t, len(c.LinearConstraint), test.ShouldEqual, 1)
+	test.That(t, c.LinearConstraint[0], test.ShouldResemble, &motionpb.LinearConstraint{
+		LineToleranceMm:          &desiredLinearTolerance,
+		OrientationToleranceDegs: &desiredOrientationTolerance,
+	})
+
+	AddOrientationConstraint(c, &desiredOrientationTolerance)
+	test.That(t, len(c.OrientationConstraint), test.ShouldEqual, 1)
+	test.That(t, c.OrientationConstraint[0], test.ShouldResemble, &motionpb.OrientationConstraint{
+		OrientationToleranceDegs: &desiredOrientationTolerance,
+	})
+
+	frameMap := map[string]string{
+		"frame1": "frame2",
+		"frame3": "frame4",
+	}
+	AddCollisionSpecification(c, frameMap)
+	test.That(t, len(c.CollisionSpecification), test.ShouldEqual, 1)
+	test.That(t, c.CollisionSpecification[0], test.ShouldResemble, &motionpb.CollisionSpecification{
+		Allows: []*motionpb.CollisionSpecification_AllowedFrameCollisions{
+			{
+				Frame1: "frame1",
+				Frame2: "frame2",
+			},
+			{
+				Frame1: "frame3",
+				Frame2: "frame4",
+			},
+		},
+	})
```

Comment:
Lets add a round-trip test at the end here.  Lets write `c` to protobuf and then immediately convert it back to the `Constraint` type by using the `ConstraintsFromProtobuf` function you wrote.  If the initial and final values are the same then we know the conversions were lossless